out/manifest.go

```diff
@@ -24,18 +24,34 @@ func NewManifest(manifestPath string) (Manifest, error) {
 	return manifest, nil
 }
 
-func (manifest *Manifest) EnvironmentVariables() map[interface{}]interface{} {
-	envVars, hasEnvVars := manifest.data["env"].(map[interface{}]interface{})
-	if !hasEnvVars {
-		envVars = make(map[interface{}]interface{})
-		manifest.data["env"] = envVars
+func (manifest *Manifest) EnvironmentVariables() []map[interface{}]interface{} {
+	apps, hasApps := manifest.data["applications"].([]interface{})
+	if !hasApps {
+		return []map[interface{}]interface{}{}
 	}
-
-	return envVars
+	appEnvVars := make([]map[interface{}]interface{}, len(apps))
+	for appIdx := range apps {
+		app, isApp := apps[appIdx].(map[interface{}]interface{})
+		if !isApp {
+			continue
+		}
+		envVars, hasEnvVars := app["env"].(map[interface{}]interface{})
+		if !hasEnvVars {
+			envVars = make(map[interface{}]interface{})
+			app["env"] = envVars
+		}
+		appEnvVars[appIdx] = envVars
+	}
+	return appEnvVars
 }
 
 func (manifest *Manifest) AddEnvironmentVariable(name, value string) {
-	manifest.EnvironmentVariables()[name] = value
+	appEnvVars := manifest.EnvironmentVariables()
+	for appIdx := range appEnvVars {
+		if appEnvVars[appIdx] != nil {
+			appEnvVars[appIdx][name] = value
```

Comment:
This feels very weird. Seems lucky that the side effect persists at all given that `.EnvironmentVariables` makes a new map every time. I guess it's relying on being able to mutate the nested maps which are also referenced within the original `manifest.data`?

Reply:
It *is* weird! Like you said, every call to `EnvironmentVariables` returns a new map, but all of those maps refer to parts of the same underlying data. The approach is basically the same as before--the original `manifest.EnvironmentVariables()[name] = value` also makes a new map and mutates its contents--but I agree that it's confusing. Maybe we could refactor for clarity so that we iterate over application environments and insert values in the same step?

Reply:
Agreed - seems like a good area to refactor.